system/db/content.go

```diff
@@ -36,6 +36,98 @@ func SetContent(target string, data url.Values) (int, error) {
 	return update(ns, id, data)
 }
 
+// UpdateContent inserts or updates values in the database.
+// Updated content will be merged into existing values from the database.
+// The `target` argument is a string made up of namespace:id (string:int)
+func UpdateContent(target string, data url.Values) (int, error) {
+	t := strings.Split(target, ":")
+	ns, id := t[0], t[1]
+
+	// check if content id == -1 (indicating new post).
+	// if so, run an insert which will assign the next auto incremented int.
+	// this is done because boltdb begins its bucket auto increment value at 0,
+	// which is the zero-value of an int in the Item struct field for ID.
+	// this is a problem when the original first post (with auto ID = 0) gets
+	// overwritten by any new post, originally having no ID, defauting to 0.
+	if id == "-1" {
```

Comment:
Yes, I agree it could make sense to remove 46-55.  I left it in there to prompt discussion of a couple of things.  

- Agree that a "-1" id should be invalid for updates, right now updateContentHandler to rejects requests with a missing ID and the handler could reject id="-1" also. 

- There really isn't much different between SetContent and  the added UpdateContent other than merging content into existing values.  I wonder if you really still need the old behavior of SetContent and the existing update function with 'replace' behavior or if the merge behavior could work for everything?  If the "-1" validation is in the handler, we might be able to eliminate some duplication here and just replace SetContent with UpdateContent...